backend/src/packages/drivers/driver.service.ts

```diff
@@ -136,9 +140,59 @@ class DriverService implements IService {
 
     const driverObject = driver.toObject();
 
+    const emailData = {
+      name: `${firstName} ${lastName}`,
+      email: email,
+      password: password,
+      signInLink: MailContent.LINK,
+    };
+
+    await mailService.sendPage(
+      { to: email, subject: MailContent.SUBJECT },
+      TemplateName.DRIVER_CREDENTIALS,
+      emailData,
+    );
+
     return { ...user, ...driverObject, group };
   }
 
+  private async generatePassword(): Promise<string> {
+    const alpha = Array.from({ length: 26 }).map((_key, index) => index + 65);
+    const alphabetUpperCase = alpha.map((x) => String.fromCodePoint(x));
+    const alphabetLowerCase = alphabetUpperCase.map((ch) => ch.toLowerCase());
+    const digits = [...Array.from({ length: 10 }).keys()].map(String);
+    const minLength = 6;
+    const maxLength = 20;
+
+    const passwordLength =
+      Math.floor(Math.random() * (maxLength - minLength + 1)) + minLength;
+
+    let password =
+      alphabetLowerCase[Math.floor(Math.random() * alphabetLowerCase.length)];
+
+    for (let index = 1; index < passwordLength; index++) {
+      const characterSet = [
+        ...alphabetUpperCase,
+        ...alphabetLowerCase,
+        ...digits,
+      ];
+      const randomCharacter = characterSet.at(
+        Math.floor(Math.random() * characterSet.length),
+      );
+      password += randomCharacter;
+    }
+
+    const hasDigit = /\d/.test(password);
+    const hasLetter = /[A-Za-z]/.test(password);
+    const hasSpaces = /^\s+$/.test(password);
+
+    if (!hasDigit || !hasLetter || hasSpaces) {
+      return await this.generatePassword();
+    }
+
+    return password;
+  }
+
```

Comment:
maybe we can use bcrypt for password generaing and then just cut it to length we need

Reply:
Sure, it's possible, but I think it's not a very good idea. At least, we should need to remove characters and we should check the password length and containing uppercase and lowcase letters and numbers. Yes, it's possible to simplify the generation method by using a string of characters and selecting from it.